src/Nethermind/Nethermind.BeaconNode/AttestationProducer.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Nethermind.Core2;
+using Nethermind.Core2.Containers;
+using Nethermind.Core2.Crypto;
+using Nethermind.Core2.Types;
+
+namespace Nethermind.BeaconNode
+{
+    public class AttestationProducer
+    {
+        private readonly BeaconChainUtility _beaconChainUtility;
+        private readonly BeaconStateAccessor _beaconStateAccessor;
+        private readonly BeaconStateTransition _beaconStateTransition;
+        private readonly ICryptographyService _cryptographyService;
+        private readonly IForkChoice _forkChoice;
+        private readonly ILogger _logger;
+        private readonly IStore _store;
+
+        public AttestationProducer(ILogger<AttestationProducer> logger,
+            ICryptographyService cryptographyService,
+            BeaconChainUtility beaconChainUtility,
+            BeaconStateAccessor beaconStateAccessor,
+            BeaconStateTransition beaconStateTransition,
+            IForkChoice forkChoice,
+            IStore store)
+        {
+            _logger = logger;
+            _cryptographyService = cryptographyService;
```

Comment:
null checks on argument s helped us many times to speed up debugging but I am fine to add them myself later in one go (since I am using resharper it should be quick)

Reply:
I have nullable checking turned on in the compiler, so the static checking will prevent most null errors from ever being possible.

There could always be calls from external code, that passes in nulls, or they can creep in other ways, so we might still want to strategically add checks, i.e. still need some runtime checking (although you may need to add the overrides to force the comparison of a non-nullable with null).

Reply:
Leave as is (nullable reference types)